### PR TITLE
fix(@desktop/keycard): can't create accounts if the main account was restored in using an existing keycard account

### PR DESCRIPTION
**Pull request by saledjenic**
Corresponding `status-go` PR:
- https://github.com/status-im/status-go/pull/2916

This PR is not a refactor (that's why in some places I've just followed the approach and code style we already had around), _**even `AddAccountModal` is very very good candidate for refactor**_. It just makes adding wallet accounts work correctly in case logged in user is a keycard user or a certain keypair we want to generate new account for is migrated to a keycard.

Fixes: #7867

### CODE REVIEW

File: ui/app/AppLayouts/Onboarding/views/KeysMainView.qml [hunk 1]
```diff
@@ -150,10 +150,10 @@ Item {
                 opacity: 0.0
             }
             // TODO: Functionality is not completed. Missing saving of keystore files which blocks wallet account creation. @see #7867
```
Comment (caybro) [inline ui/app/AppLayouts/Onboarding/views/KeysMainView.qml:152]: TODO still valid?

File: ui/app/AppLayouts/Wallet/panels/DerivedAddressesPanel.qml [hunk 2]
```diff
@@ -117,6 +138,18 @@ Item {
...
+
+        StatusButton {
+            visible: _internal.showEnterPinPassButton
+            text: derivedAddresses.selectedKeyUidMigratedToKeycard || userProfile.isKeycardUser?
+                      qsTr("Enter PIN") :
+                      qsTr("Enter password")
+            icon.name: derivedAddresses.enterPasswordIcon
+            highlighted: focus
+
+            Keys.onReturnPressed: _internal.runAction()
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/panels/DerivedAddressesPanel.qml:150]: How can you press Return on a button here? :)

File: ui/app/AppLayouts/Wallet/popups/AddAccountModal.qml [hunk 3]
```diff
@@ -214,6 +252,7 @@ StatusModal {
             StatusColorSelectorGrid {
                 id: colorSelectionGrid
                 anchors.horizontalCenter: parent.horizontalCenter
+                enabled: accountNameInput.valid
                 titleText: qsTr("color").toUpperCase()
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/AddAccountModal.qml:256]: :D

File: ui/app/AppLayouts/Wallet/panels/DerivedAddressesPanel.qml (hunk 2 above)
Comment (saledjenic) [reply ui/app/AppLayouts/Wallet/panels/DerivedAddressesPanel.qml:150]: @caybro that's the consequence of copy/paste (I copied button from `AddAccountModal.qml`), that's why I said what I said in the description of this PR. :)

File: ui/app/AppLayouts/Wallet/popups/AddAccountModal.qml (hunk 3 above)
Comment (saledjenic) [reply ui/app/AppLayouts/Wallet/popups/AddAccountModal.qml:256]: @caybro hahahaha that's the best practice thing ;)

File: ui/app/AppLayouts/Onboarding/views/KeysMainView.qml (hunk 1 above)
Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/views/KeysMainView.qml:152]: @caybro no, it isn't, forgot to remove it, removed now.

**Comment by glitchminer** [comment]
@saledjenic, signup with keycard appears to be broken so couldn't reach point of restoring then to test creating accounts

- Generate keys for keycard
- Select I prefer to use pin (error shown)
- Select I prefer to use pin (different error shown)

image
image

**Comment by saledjenic** [comment]
@glitchminer this is something very weird, I haven't seen that while I was testing and tested it one more time just now and went well. Could you give me detailed steps what you did? Also if you can share the log would be great, cause for the develop build there should be all keycard steps logged so I can reconstruct what you did.

**Review by glitchminer — APPROVED** [review]
@saledjenic, seems to have been due to conflict due to multiple instances. Retested now and passed.

Test steps:
Restore account from keycard
Enable wallet
Add second account using default as origin
Add third account from importing seed phrase as origin

**Review by alaibe — APPROVED** [review]

**Review by stefandunca — APPROVED** [review]
Landing a hand with an instaprove

**Review by borismelnik — APPROVED** [review]
LGTM :)

**Pull request merged**